engine/gl/gl_heightmap.c

```diff
@@ -5738,7 +5738,7 @@ void Terr_Brush_Draw(heightmap_t *hm, batch_t **batches, entity_t *e)
 		{
 			if (br->faces[j].relight && dorelight)
 			{
-				lightstyleindex_t styles[4] = {0,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE};
+				lightstyleindex_t styles[MAXCPULIGHTMAPS] = {0,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE,INVALID_LIGHTSTYLE};
```

Comment:
ops.. perhaps should match size, or perhaps drop the last two ones there as it breaks iteration on first invalid.